src/gf.c

```diff
@@ -126,7 +126,7 @@ const struct jl_typemap_info method_defs = {
     0, &jl_method_type
 };
 const struct jl_typemap_info lambda_cache = {
-    0, &jl_method_instance_type
+    1, &jl_method_instance_type
```

Comment:
Ok. but it surely doesn't require sorting in the same sense --- this doesn't cause any test failures. We might need another flag telling the typemap that only simple signatures will be inserted (i.e. the stuff we put in method caches, basically leaf types and `Any`s).